lib/rules/container-name-pattern/index.mjs

```diff
@@ -0,0 +1,105 @@
+import valueParser from 'postcss-value-parser';
+
+import { atRuleRegexes, propertyRegexes } from '../../utils/regexes.mjs';
+import { isRegExp, isString } from '../../utils/validateTypes.mjs';
+import { atRuleParamIndex } from '../../utils/nodeFieldIndices.mjs';
+import { declarationValueIndex } from '../../utils/nodeFieldIndices.mjs';
+import isStandardSyntaxAtRule from '../../utils/isStandardSyntaxAtRule.mjs';
+import isStandardSyntaxProperty from '../../utils/isStandardSyntaxProperty.mjs';
+import report from '../../utils/report.mjs';
+import ruleMessages from '../../utils/ruleMessages.mjs';
+import validateOptions from '../../utils/validateOptions.mjs';
+
+const ruleName = 'container-name-pattern';
+
+const messages = ruleMessages(ruleName, {
+	expected: (containerName, pattern) => `Expected "${containerName}" to match pattern "${pattern}"`,
+});
+
+const meta = {
+	url: 'https://stylelint.io/user-guide/rules/container-name-pattern',
+};
+
+const NON_CONTAINER_NAMES = new Set(['and', 'or', 'none', 'not']);
```

Comment:
```suggestion
const KEYWORDS = new Set(['and', 'or', 'none', 'not']);
```

Let's rename this var as these are keywords, whereas the container names are custom-idents.

See [specs](https://drafts.csswg.org/css-conditional-5/#container-name).